substrate/frame/support/src/storage/storage_noop_guard.rs

```diff
@@ -37,11 +37,40 @@
 /// });
 /// ```
 #[must_use]
-pub struct StorageNoopGuard(sp_std::vec::Vec<u8>);
+pub struct StorageNoopGuard {
+	storage_root: sp_std::vec::Vec<u8>,
+	error_message: &'static str,
+}
 
 impl Default for StorageNoopGuard {
 	fn default() -> Self {
-		Self(sp_io::storage::root(sp_runtime::StateVersion::V1))
+		Self {
+			storage_root: sp_io::storage::root(sp_runtime::StateVersion::V1),
+			error_message: "`StorageNoopGuard` detected an attempted storage change.",
+		}
+	}
+}
+
+impl From<&'static str> for StorageNoopGuard {
+	fn from(error_message: &'static str) -> Self {
+		StorageNoopGuard { storage_root: sp_std::vec::Vec::new(), error_message }
```

Comment:
did you mean:
```suggestion
		StorageNoopGuard { storage_root: sp_io::storage::root(sp_runtime::StateVersion::V1), error_message }
```

Reply:
Oops, brainfart. I've removed this in the end though after the comment from @ggwpez which changed how I thought about it.